### DISCUSSION TITLE
db errors probably due to wrong permissions

### DISCUSSION THREAD

**Discussion by steakhutzeee** [firefly-iii/firefly-iii#4351] [Firefly III] [answered]
Hi, new to Firefly and I really like this project.

Setup with docker compose with basically all default settings, but I have some errors in the db logs.

`
fireflyiii_db

    yesterday at 16:19  [i] mysqld not found, creating....
    yesterday at 16:19  [i] MySQL data directory not found, creating initial DBs
    yesterday at 16:19  [i] MySQL root Password: hidden
    yesterday at 16:19  [i] Creating database: firefly
    yesterday at 16:19  [i] with character set: 'utf8' and collation: 'utf8_general_ci'
    yesterday at 16:19  [i] Creating user: firefly with password secret_firefly_password
    yesterday at 16:19  2021-02-04 15:19:47 0 [Note] /usr/bin/mysqld (mysqld 10.4.15-MariaDB) starting as process 73 ...
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: Using Linux native AIO
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: Uses event mutexes
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: Number of pools: 1
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: Using generic crc32 instructions
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] mysqld: O_TMPFILE is not supported on /var/tmp (disabling future attempts)
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: Completed initialization of buffer pool
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: 128 out of 128 rollback segments are active.
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: Creating shared tablespace for temporary tables
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: 10.4.15 started; log sequence number 61090; transaction id 21
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] Plugin 'FEEDBACK' is disabled.
    yesterday at 16:19  2021-02-04 15:19:48 0 [Note] InnoDB: Buffer pool(s) load completed at 210204 15:19:48
    yesterday at 16:19  2021-02-04 15:19:48 6 [Warning] 'user' entry '@04d1ecfd882d' ignored in --skip-name-resolve mode.
    yesterday at 16:19  2021-02-04 15:19:48 6 [Warning] 'proxies_priv' entry '@% root@04d1ecfd882d' ignored in --skip-name-resolve mode.
    yesterday at 16:19  2021-02-04 15:19:48 6 [Warning] 'user' entry '@04d1ecfd882d' ignored in --skip-name-resolve mode.
    yesterday at 16:19  2021-02-04 15:19:48 6 [Warning] 'proxies_priv' entry '@% root@04d1ecfd882d' ignored in --skip-name-resolve mode.
    yesterday at 16:19  /scripts/run.sh: ignoring or entrypoint initdb empty /docker-entrypoint-initdb.d/*
    yesterday at 16:19  
    yesterday at 16:19  
    yesterday at 16:19  MySQL init process done. Ready for start up.
    yesterday at 16:19  
    yesterday at 16:19  exec /usr/bin/mysqld --user=mysql --console --skip-name-resolve --skip-networking=0
    yesterday at 16:19  2021-02-04 15:19:50 0 [Note] /usr/bin/mysqld (mysqld 10.4.15-MariaDB) starting as process 1 ...
    yesterday at 16:19  2021-02-04 15:19:50 0 [Note] InnoDB: Using Linux native AIO
    yesterday at 16:19  2021-02-04 15:19:50 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
    yesterday at 16:19  2021-02-04 15:19:50 0 [Note] InnoDB: Uses event mutexes
    yesterday at 16:19  2021-02-04 15:19:50 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
    yesterday at 16:19  2021-02-04 15:19:50 0 [Note] InnoDB: Number of pools: 1
    yesterday at 16:19  2021-02-04 15:19:50 0 [Note] InnoDB: Using generic crc32 instructions
    yesterday at 16:19  2021-02-04 15:19:50 0 [Note] mysqld: O_TMPFILE is not supported on /var/tmp (disabling future attempts)
    yesterday at 16:19  2021-02-04 15:19:50 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
    yesterday at 16:19  2021-02-04 15:19:50 0 [Note] InnoDB: Completed initialization of buffer pool
    yesterday at 16:19  2021-02-04 15:19:50 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
    yesterday at 16:19  2021-02-04 15:19:51 0 [Note] InnoDB: 128 out of 128 rollback segments are active.
    yesterday at 16:19  2021-02-04 15:19:51 0 [Note] InnoDB: Creating shared tablespace for temporary tables
    yesterday at 16:19  2021-02-04 15:19:51 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
    yesterday at 16:19  2021-02-04 15:19:51 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
    yesterday at 16:19  2021-02-04 15:19:51 0 [Note] InnoDB: 10.4.15 started; log sequence number 61099; transaction id 21
    yesterday at 16:19  2021-02-04 15:19:51 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
    yesterday at 16:19  2021-02-04 15:19:51 0 [Note] Plugin 'FEEDBACK' is disabled.
    yesterday at 16:19  2021-02-04 15:19:51 0 [Note] InnoDB: Buffer pool(s) load completed at 210204 15:19:51
    yesterday at 16:19  2021-02-04 15:19:51 0 [Note] Server socket created on IP: '::'.
    yesterday at 16:19  2021-02-04 15:19:51 0 [Warning] 'user' entry '@04d1ecfd882d' ignored in --skip-name-resolve mode.
    yesterday at 16:19  2021-02-04 15:19:51 0 [Warning] 'proxies_priv' entry '@% root@04d1ecfd882d' ignored in --skip-name-resolve mode.
    yesterday at 16:19  2021-02-04 15:19:51 0 [Note] Reading of all Master_info entries succeeded
    yesterday at 16:19  2021-02-04 15:19:51 0 [Note] Added new Master_info '' to hash table
    yesterday at 16:19  2021-02-04 15:19:51 0 [Note] /usr/bin/mysqld: ready for connections.
    yesterday at 16:19  Version: '10.4.15-MariaDB'  socket: '/run/mysqld/mysqld.sock'  port: 3306  MariaDB Server
    yesterday at 16:19  2021-02-04 15:19:51 8 [Warning] Aborted connection 8 to db: 'unconnected' user: 'unauthenticated' host: '172.18.0.16' (This connection closed normally without authentication)
    yesterday at 16:50  2021-02-04 15:50:35 0 [Note] /usr/bin/mysqld (initiated by: unknown): Normal shutdown
    yesterday at 16:50  2021-02-04 15:50:35 0 [Note] Event Scheduler: Purging the queue. 0 events
    yesterday at 16:50  2021-02-04 15:50:35 0 [Note] InnoDB: FTS optimize thread exiting.
    yesterday at 16:50  2021-02-04 15:50:35 0 [Note] InnoDB: Starting shutdown...
    yesterday at 16:50  2021-02-04 15:50:35 0 [ERROR] InnoDB: Operating system error number 13 in a file operation.
    yesterday at 16:50  2021-02-04 15:50:35 0 [ERROR] InnoDB: The error means mysqld does not have the access rights to the directory.
    yesterday at 16:50  2021-02-04 15:50:35 0 [Note] InnoDB: Dumping buffer pool(s) to /var/lib/mysql/ib_buffer_pool
    yesterday at 16:50  2021-02-04 15:50:35 0 [ERROR] InnoDB: Cannot open '/var/lib/mysql/ib_buffer_pool.incomplete' for writing: Permission denied
    yesterday at 16:50  2021-02-04 15:50:37 0 [Note] InnoDB: Shutdown completed; log sequence number 3265698; transaction id 3624
    yesterday at 16:50  2021-02-04 15:50:37 0 [ERROR] InnoDB: Operating system error number 13 in a file operation.
    yesterday at 16:50  2021-02-04 15:50:37 0 [ERROR] InnoDB: The error means mysqld does not have the access rights to the directory.
    yesterday at 16:50  2021-02-04 15:50:37 0 [Note] /usr/bin/mysqld: Shutdown complete
    yesterday at 16:50  
    yesterday at 16:50  [i] mysqld already present, skipping creation
    yesterday at 16:50  [i] MySQL directory already present, skipping creation
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] /usr/bin/mysqld (mysqld 10.4.15-MariaDB) starting as process 1 ...
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: Using Linux native AIO
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: Uses event mutexes
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: Number of pools: 1
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: Using generic crc32 instructions
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] mysqld: O_TMPFILE is not supported on /var/tmp (disabling future attempts)
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: Completed initialization of buffer pool
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: 128 out of 128 rollback segments are active.
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: Removed temporary tablespace data file: "ibtmp1"
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: Creating shared tablespace for temporary tables
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: 10.4.15 started; log sequence number 3265698; transaction id 3626
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] Plugin 'FEEDBACK' is disabled.
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] InnoDB: Buffer pool(s) load completed at 210204 15:50:42
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] Server socket created on IP: '::'.
    yesterday at 16:50  2021-02-04 15:50:42 0 [Warning] 'user' entry '@04d1ecfd882d' ignored in --skip-name-resolve mode.
    yesterday at 16:50  2021-02-04 15:50:42 0 [Warning] 'proxies_priv' entry '@% root@04d1ecfd882d' ignored in --skip-name-resolve mode.
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] Reading of all Master_info entries succeeded
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] Added new Master_info '' to hash table
    yesterday at 16:50  2021-02-04 15:50:42 0 [Note] /usr/bin/mysqld: ready for connections.
    yesterday at 16:50  Version: '10.4.15-MariaDB'  socket: '/run/mysqld/mysqld.sock'  port: 3306  MariaDB Server
    yesterday at 16:50  2021-02-04 15:50:45 8 [Warning] Aborted connection 8 to db: 'unconnected' user: 'unauthenticated' host: '172.18.0.16' (This connection closed normally without authentication)
`

Tried to fix permissions to my db and upload dirs but issue persists. 

`drwxrws---+  4 www-data     www-data 4096 Feb  4 16:15 'Firefly III'`

What could be the issue? 

I usually use containers by linuxserver with PUID and PGID of my user. Can I do the same here? 

Thank you in advance.

**Comment by steakhutzeee** [comment]
Checking permissions for mysql and upload dirs:

`root@DK:/srv/dev-disk-by-label-HC2/AppData/Firefly III# ls -l
total 8
drwxrws---+ 5 _apt     systemd-journal 4096 Feb  5 23:34 mysql
drwxrwsr-x+ 2 www-data www-data        4096 Feb  4 16:14 upload`

**Comment by steakhutzeee** [comment, marked as answer]
Actually i moved to another MariaDB image and had no issues [https://hub.docker.com/r/linuxserver/mariadb](https://hub.docker.com/r/linuxserver/mariadb)

**Reply by JC5** [reply]
Thanks for letting me know. I was looking for a good DB alternative. Should've known linuxserver has mariadb lol
